src/app/app.component.ts

```diff
@@ -3,23 +3,56 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-import { Component, OnInit } from '@angular/core';
+import { Component, Inject, OnInit } from '@angular/core'
+import { ShareLink } from './shareUrl/sharelink-url'
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
 export class AppComponent implements OnInit {
+  readonly TRIPLE_BANANA_DEV_URL = "https://triplebanana.dev"
+
   readonly DEFAULT_BANNER_DESCRIPTION: string = "Safer, faster, and more stable"
+  readonly ANDROID_CHROME_CLICK_DESCRIPTION = "Click to open link with Banana browser"
+
+  readonly CLICK_GUIDE_WAITING_IN_MS = 500
+
   bannerDescription = this.DEFAULT_BANNER_DESCRIPTION
 
+  private shareLink: ShareLink
+
+  constructor(
+    @Inject('Location') private location: Location
+  ) {
+    this.shareLink = this.getShareLinkUrl()
```

Comment:
Q. which is better place to call `getShareLinkUrl`?
`constructor` or `ngOnInit`

Reply:
For working, no matter where to place constructor or ngOnInit, but reason why I called it on constructor, the current url of this site which contain share url is already decided and should it be separated from angular f/w so call it as soon as possible before processing in angular f/w's code